extensions/arc/deployment/src/main/resources/dev-ui/arc/qwc-arc-beans-dependency.js

```diff
@@ -0,0 +1,98 @@
+import { LitElement, html, css} from 'lit';
+import { arcState } from './arc-state.js';
+import { dependencyGraphs } from 'arc-data';
+import 'qui-d3-force';
+
+/**
+ * This component shows the Arc Beans Dependency
+ */
+export class QwcArcBeansDependency extends LitElement {
+    
+    static styles = css`
+        .d3Container {
+            width: 100%;
+            height: 100%;
+            max-width: 100%;
+            max-height: 100%;
+        }
+    `;
+
+    static properties = {
+        _heading: {state: true},
+        _nodes: {state: true},
+        _links: {state: true},
+    };
+
+    constructor() {
+        super();
+
+        this._refresh();
+
+        this._types = [
+            { name: 'root', icon: 'font-awesome-solid:circle', description: 'root', color: 'var(--lumo-error-text-color)'},
+            { name: 'leaf', icon: 'font-awesome-solid:circle', color: 'var(--lumo-contrast)'},
+            { name: 'directDependency', icon: 'font-awesome-solid:arrow-right', description: 'direct dependencies', color: '#1f77b4'},
+            { name: 'directDependent', icon: 'font-awesome-solid:arrow-right', description: 'direct dependents', color: '#ff7f0e'},
+            { name: 'dependency', icon: 'font-awesome-solid:arrow-right', description: 'dependencies', color: '#2ca02c'},
+            { name: 'lookup', icon: 'font-awesome-solid:arrow-right', description: 'potential dependency', subtext: 'programmatic lookup', color: '#d62728'},
+            { name: 'producer', icon: 'font-awesome-solid:arrow-right', description: 'declaring bean of a producer', color: '#9467bd'}
+        ];
+    }
+
+    render() {
+        return html`
+        <div class="d3Container" >
+            <qui-d3-force  
+                @code-clicked="${(e) => { this._nodeClicked(e) }}" 
+                @back-clicked="${() => { this._backClicked() }}" 
+                .nodes=${this._nodes}
+                .links=${this._links}
+                .types=${this._types}
+                .title=${this._heading}>
+            </qui-d3-force>
+        </div>`;
+    }
+
+    _refresh(){
+        let d = dependencyGraphs[arcState.beanId];
+
+        this._nodes = [];
+        this._links = [];
+        
+        d.nodes.forEach(n => {
+            let type = "leaf";
+            if(n.id === arcState.beanId){
+                type = "root";        
+                this._heading = n.providerType.name;
```

Comment:
`n.providerType.name` should be replaced with `DevBeanInfo.description(false)` (which is a computed property and is probably not serialized in the `ArcDevUIProcessor`?).

`providerType` is not acceptable for producers, e.g. the new Dev UI currently shows `java.lang.String` instead of `io.smallrye.config.inject.ConfigProducer#produceStringConfigProperty()`.